src/extension/webview.ts

```diff
@@ -0,0 +1,40 @@
+import {
+    Uri,
+    ViewColumn,
+    WebviewPanel,
+    window,
+    workspace,
+} from "vscode";
+
+export class PreviewPanel {
+    private pickedReport: string;
+    private previewPanel: WebviewPanel;
+
+    constructor(pickedReport: string) {
+        this.pickedReport = pickedReport;
+        // Construct the webview panel for the coverage report to live in
+        this.previewPanel = window.createWebviewPanel(
+            "coverageReportPreview",
+            "Preview Coverage Report",
+            ViewColumn.One,
+        );
+    }
+
+    public async createWebView() {
+        // Read in the report html and send it to the webview
+        const reportUri = Uri.file(this.pickedReport);
+        const reportHtml = await workspace.openTextDocument(reportUri);
+        const reportHtmlWithPolicy = this.addContentSecurityPolicy(reportHtml.getText());
+
+        this.previewPanel.webview.html = reportHtmlWithPolicy;
+    }
+
+    private addContentSecurityPolicy(text: string) {
```

Comment:
minor: we could add a return type of string here I guess 🤔

Reply:
Ok. Will do that